sklearn/model_selection/_search.py

```diff
@@ -994,3 +999,356 @@ def fit(self, X, y=None, labels=None):
+
+    def _evaluate_params(self, params, cv, X, y, labels):
+        all_scores = Parallel(
+            n_jobs=self.n_jobs, verbose=self.verbose,
+            pre_dispatch=self.pre_dispatch
+        )(delayed(_fit_and_score)(clone(self.estimator), X, y, self.scorer_,
+                                  train, test, self.verbose, params,
+                                  self.fit_params, return_parameters=False,
+                                  error_score=self.error_score)
+          for train, test in cv.split(X, y, labels))
+
+        n_test_samples = 0
+        score = 0
+        for tmp_score, tmp_n_test_samples, _ in all_scores:
+            tmp_score *= tmp_n_test_samples
+            n_test_samples += tmp_n_test_samples
+            score += tmp_score
+        score /= float(n_test_samples)
+
+        return score, all_scores
+
+    def fit(self, X, y=None, labels=None):
+        """
+        Run the hyper-parameter optimization process
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vector, where n_samples in the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output], optional
+            Target relative to X for classification or regression;
+            None for unsupervised learning.
+
+        labels : array-like, with shape (n_samples,), optional
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+        """
+        cv_scores, grid_scores, tested_parameters = [], [], []
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+        cv = check_cv(self.cv, X, classifier=is_classifier(self.estimator))
+
+        #  Initialize with random candidates
+        init_candidates = ParameterSampler(
+            self.param_distributions, self.n_init,
+            random_state=self.random_state)
+
+        for i, params in enumerate(init_candidates):
+            score, all_scores = self._evaluate_params(
+                params, cv, X, y, labels)
+
+            if self.verbose > 0:
+                print('Step ' + str(i) + ' - Hyperparameter '
+                      + str(params) + ', score: ' + str(score))
+            tested_parameters.append(list(params.values()))
+            cv_scores.append(score)
+            grid_scores.append(
+                _CVScoreTuple(params, score, np.array(all_scores)))
+
+        for i in range(self.n_iter-self.n_init):
+            # Model with a Gaussian Process
+            gp = GaussianProcessRegressor(**self.gp_params)
+            gp.fit(tested_parameters, cv_scores)
+
+            # Sample candidates and predict their corresponding
+            # acquisition values
+            candidates = ParameterSampler(
+                self.param_distributions, self.n_candidates,
+                random_state=self.random_state)
```

Comment:
you should instead initialize `random_state = check_random_state(self.random_state)` at the beginning of `fit`, otherwise candidates sampled here will be identical at each iteration.